java/arcs/core/entity/EntityBase.kt

```diff
@@ -395,6 +398,28 @@ private fun fromReferencable(
             requireNotNull(referencable.value) {
                 "ReferencableList encoded an unexpected null value."
             }
+            val listType = type.primitiveType
+            when (listType) {
+                is FieldType.Primitive -> {
+                    referencable.value.map {
+                        require(it is ReferencablePrimitive<*>) {
+                            "Expected List to contain ReferencablePrimitives but found $it."
+                        }
+                        it.value
```

Comment:
could this be fromReferencable(it, listType, nestedEntitySpecs) to reduce code duplication?

Reply:
Ah, yes this is much nicer